Makefile

```diff
@@ -1,37 +1,40 @@
 DEPS = deps/meck
 
-all: compile eunit dialyzer
-
-clean:
-	@rebar skip_deps=true clean
+REBAR=$(shell which rebar3 || echo ./rebar3)
 
-allclean:
-	@rebar clean
+all: compile eunit dialyzer
 
-distclean:
-	@rebar skip_deps=true clean
-	@rm -rf deps
+clean: $(REBAR)
+	$(REBAR) as dev clean
+	$(REBAR) as test clean
 
-compile: $(DEPS)
-	@rebar compile
+compile: $(REBAR) $(DEPS)
+	$(REBAR) as dev compile
 
 dialyzer:
-	@dialyzer -Wunderspecs -r ebin
+	$(REBAR) dialyzer
 
-$(DEPS):
-	@rebar get-deps
+$(DEPS): $(REBAR)
+	$(REBAR) get-deps
```

Comment:
I think with rebar3 we shouldn't really need this target anymore, but it is fine to keep around